### PR TITLE
Support for Doctrine ORM >=2.5

### PR DESCRIPTION
**Pull request by pamil**

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
Could you update other composer.json's?

https://github.com/Sylius/Sylius/search?l=json&q=doctrine%5C%2Form&type=Code&utf8=%E2%9C%93

**Comment by michalmarcinkowski** [comment]
@pamil does the fix work on `v2.5.1`? If not we should exclude this version.

File: src/Sylius/Component/Channel/composer.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "require": {
         "php": ">=5.5.9",
 
-        "doctrine/orm": "2.4.*",
+        "doctrine/orm": "^2.4.8",
```
Comment (pjedrzejewski) [inline src/Sylius/Component/Channel/composer.json]: Channel really requires Doctrine ORM? Should not be the case ...

**Comment by pamil** [comment]
@michalmarcinkowski it should be fine on all `2.5.*` releases.

File: src/Sylius/Component/Channel/composer.json (hunk 1 above)
Comment (pamil) [inline src/Sylius/Component/Channel/composer.json]: Looks like ORM's exceptions are used in PhpDocs in `ChannelRepositoryInterface`.

`doctrine/common` is used too in `ChannelsAwareInterface` only.

Comment (michalmarcinkowski) [inline src/Sylius/Component/Channel/composer.json]: ORM's exceptions should be removed from interface docs. The interface doesn't know anything about the implementation especially about the thrown exceptions.

Let's require `doctrine/common` then.

**Comment by pjedrzejewski** [comment]
Thanks Kamil!

File: src/Sylius/Component/Channel/composer.json (hunk 1 above)
Comment (pjedrzejewski) [inline src/Sylius/Component/Channel/composer.json]: I opened a separate issue for that #3729.

Comment (michalmarcinkowski) [inline src/Sylius/Component/Channel/composer.json]: :+1:

**Pull request merged**